Training/Program.cs

```diff
@@ -1,33 +1,82 @@
 
-      #region Method ------------------------------------------------
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
-      }
-      #endregion
-
-      #region Operators ---------------------------------------------
-      #endregion
+#region class Program -----------------------------------------------------------------------------
+internal class Program {
+   #region Method ---------------------------------------------------
+   static void Main (string[] args) {
+      Write ("input:");
+      string input = ReadLine ().Trim ().ToLower () ?? "";
+      bool isValid = DoubleParse.Parse (input, out double num);
+      WriteLine ("output: " + num + $"\n{(isValid ? "" : "invalid input")}");
+   }
+   #endregion
+}
+#endregion
 
-      #region Implementation ----------------------------------------
-      #endregion
+#region Class DoubleParse -------------------------------------------------------------------------
+public class DoubleParse {
+   #region Method ---------------------------------------------------
+   /// <summary>Converts string to double.</summary>
+   /// <param name="input">string input</param>
+   /// <param name="num">double number</param>
+   /// <returns>If False is input number is invalid, true otherwise.</returns>
+   public static bool Parse (string input, out double num) {
+      num = 0;
+      if (input != "") {
+         string[] number = input.Split ('e');
+         if (number.Length > 2) return false;
+         if (Validate (number[0] ?? "", out bool isNegative)) {
+            num = Digits (number[0], isNegative);
+            if (number.Length == 2)
+               if (Validate (number[1], out isNegative)) {
+                  double exponent = Digits (number[1], isNegative);
+                  num *= Math.Pow (10, exponent);
+               } else {
+                  num = 0;
+                  return false;
+               }
+            return true;
+         }
+      }
+      return false;
+   }
+   #endregion
 
-      #region Nested Types ------------------------------------------
-      #endregion
+   #region Implementation -------------------------------------------
+   /// <summary>Returns double of input string. Removes sign from number and splits it into integer
+   /// and fractional parts which are converted to double after matching from a dictionary <char, int></summary>
+   static double Digits (string input, bool isNegative = false) {
+      Dictionary<char, int> numbers = new () { { '0', 0 }, { '1', 1 }, { '2', 2 }, { '3', 3 }, { '4', 4},
+                                               { '5', 5 }, { '6', 6 }, {'7', 7 }, {'8', 8 }, {'9', 9 } };
+      string[] parts = (input[0] is '-' or '+' ? input[1..input.Length] : input).Split ('.');
+      (double num, int intLength) = (0, parts[0].Length - 1);
+      foreach (char c in parts.Length == 1 ? parts[0] : parts[0] + parts[1])
+         num += numbers[c] * Math.Pow (10, intLength--);

```

Comment:
For clarity, move ternary expression to a new variable.
```suggestion
     string str = parts.Length == 1 ? parts[0] : parts[0] + parts[1];
      foreach (char c in str)
         num += numbers[c] * Math.Pow (10, intLength--);

```

Reply:
Also, add comments as to why you have two case parts[0] will have digits before dot and parts[1] will have digits after dot. This will help us to understand why intLength should be parts[0].Length.

Reply:
Yes. Summary should tell us what the method will do. When reading the code, we will read the line and if variable name is  generic, you have to explain here.